frontend/src/scenes/organization/Settings/DangerZone.tsx

```diff
@@ -28,15 +29,32 @@ export function DeleteOrganizationModal({
                 // @ts-expect-error - data-attr works just fine despite not being in ButtonProps
                 'data-attr': 'delete-organization-ok',
                 loading: isDeletionInProgress,
+                disabled: !isDeletionConfirmed,
             }}
             onCancel={() => setIsVisible(false)}
             cancelButtonProps={{
                 disabled: isDeletionInProgress,
             }}
             visible={isVisible}
         >
-            Organization deletion <b>cannot be undone</b>. You will lose all data, <b>including all events</b>, related
-            to all projects within this organization.
+            <p>
+                Organization deletion <b>cannot be undone</b>. You will lose all data, <b>including all events</b>,
+                related to all projects within this organization.
+            </p>
+            <p>
+                Please type{' '}
+                <strong>{currentOrganization ? currentOrganization.name : "this organization's name"}</strong> to
+                confirm.
+            </p>
+            <Input
+                type="text"
+                onChange={(e) => {
+                    if (currentOrganization) {
+                        const { value } = e.target
+                        setIsDeletionConfirmed(value.toLowerCase() === currentOrganization.name.toLowerCase())
```

Comment:
What happens if the project is `null` here? You have 'this project's name' above, but in that case, the name will be `null` right?

Reply:
`"this project's name"` is only for cleaner code, it should never be seen in operation really (or only for a split second)

Reply:
Makes sense 👍